### PR TITLE
Chore/rm safemath

### PR DESCRIPTION
**Pull request by refcell**
## Chore: Remove SafeMath

Removes OpenZeppelin's SafeMath Library.

## Why

As of Solidity >=0.8.0, Checked Arithmetic is completed at the language level. See the Solidity [0.8.0 Release Announcement](https://blog.soliditylang.org/2020/12/16/solidity-v0.8.0-release-announcement/).

### CODE REVIEW

File: src/Rental.sol [hunk 1]
```diff
@@ -179,8 +174,8 @@ contract Rental {
         }
 
         // If the borrower sent too much ETH, immediately refund them the extra ETH they sent 
-        if (msg.value > rentalPayment.add(collateral)) {
-            payable(msg.sender).transfer(msg.value.sub(rentalPayment.add(collateral)));
+        if (msg.value > rentalPayment + collateral) {
+            payable(msg.sender).transfer(msg.value - (rentalPayment - collateral));
```
Comment (GrantStenger) [inline src/Rental.sol]: The final minus should be flipped here. Here are two options. 

payable(msg.sender).transfer(msg.value - (rentalPayment + collateral));

payable(msg.sender).transfer(msg.value - rentalPayment - collateral);

Which style here is up to you.

File: src/Rental.sol [hunk 2]
```diff
@@ -133,7 +128,7 @@ contract Rental {
 
     // After the contract is constructed with the parameters informally agreed upon off-chain,
     // the lender must deposit the designated NFT if they want to receive the rental payment.
-    // The lender 
+    // The lender
```
Comment (GrantStenger) [inline src/Rental.sol]: We can remove this line completely.

File: src/Rental.sol [hunk 3]
```diff
@@ -287,11 +282,11 @@ contract Rental {
         uint256 withdrawableCollateral;
         uint256 timeLeftUntilFullyPaid = rentalStartTime + collateralPayoutPeriod - block.timestamp;
         if (timeLeftUntilFullyPaid > 0) {
-            withdrawableCollateral = address(this).balance.sub(collateral.mul(timeLeftUntilFullyPaid.div(collateralPayoutPeriod)));
+            withdrawableCollateral = address(this).balance - (collateral * timeLeftUntilFullyPaid) / collateralPayoutPeriod;
```
Comment (GrantStenger) [inline src/Rental.sol]: I think the parentheses around collateral and timeLeftUntilFullyPaid can be removed. 

withdrawableCollateral = address(this).balance - collateral * timeLeftUntilFullyPaid / collateralPayoutPeriod;

**Review by GrantStenger — COMMENTED** [review]
Thanks for making these changes. I've added my comments for what needs to be tweaked here. I also want to fully understand what the consequences of not using safemath are. In normal scenarios, adding collateral and the rental payment have no chance of overflowing. If the contract is created with an absurdly large collateralPayoutPeriod or other malicious parameters, the counter-party wouldn't accept the offer in the first place, so as long as users are checking the terms of the contract before interacting with it we likely don't need safemath.

File: src/Rental.sol (hunk 1 above)
Comment (refcell) [reply src/Rental.sol]: oops good catch :pray:

File: src/Rental.sol (hunk 3 above)
Comment (refcell) [reply src/Rental.sol]: ya totally, can do - just left for verbosity

**Review by GrantStenger — APPROVED** [review]
Looks great, nice work!

**Pull request merged**